lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,219 @@
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::MemberProductionCounts;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::Utils::SqlHelper;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'MemberProductionCounts',
+  DESCRIPTION    => 'Checks that the gene_member counts are appropriately populated',
+  GROUPS         => ['compara', 'compara_protein_trees'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['CAFE_gene_family', 'family', 'gene_member', 'gene_member_hom_stats', 'gene_tree_root', 'genome_db']
+};
+
+sub tests {
+  my ($self) = @_;
+  my $dba = $self->dba;
+  my $dbc = $dba->dbc;
+  my $helper = $dbc->sql_helper;
+  my $gdb_adap = $dba->get_GenomeDBAdaptor;
+  
+  fk($dbc, "gene_member_hom_stats", "gene_member_id", "gene_member", "gene_member_id");
+  my $constraint = "tree_type = 'tree' AND ref_root_id IS NULL";
+  fk($dbc, "gene_member_hom_stats", "collection", "gene_tree_root", "clusterset_id", $constraint);
+
+  my $sqlCollections = qq/
+     SELECT DISTINCT clusterset_id 
+       FROM gene_tree_root 
+     WHERE $constraint
+   /;
+
+  my $collections = $helper->execute_simple( -SQL => $sqlCollections );
+
+  my @columns = qw(families gene_trees gene_gain_loss_trees orthologues paralogues homoeologues);
+
+  foreach my $collection ( @$collections ) {
+    my $sql_sums = qq/
+      SELECT SUM(families) AS sum_families, SUM(gene_trees) AS sum_gene_trees, SUM(gene_gain_loss_trees) AS sum_gene_gain_loss_trees, SUM(orthologues) AS sum_orthologues, SUM(paralogues) AS sum_paralogues, SUM(homoeologues) AS sum_homoeologues 
+        FROM gene_member_hom_stats 
+      WHERE collection = "$collection"
+    /;
+
+    my $sums = $helper->execute( -SQL => $sql_sums, -USE_HASHREFS => 1 );
+
+    #First check counts are > 0 for the collections from sql queries
+    my $sqlFamilies = q/
+      SELECT COUNT(*) 
+        FROM family
+    /; # To be tested only with default
+    my $sqlGeneTrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+      WHERE clusterset_id = "$collection"
+    /;
+    my $sqlPolyploids = q/
+      SELECT COUNT(*) 
+        FROM genome_db 
+      WHERE genome_component IS NOT NULL
+    /; # Assumes that the polyploid genomes are found in all the collections
+    my $sqlCAFETrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+          JOIN CAFE_gene_family 
+            ON gene_tree_root.root_id = gene_tree_root_id 
+      WHERE clusterset_id = "$collection"
+    /;
+
+    my $family_count = $helper->execute_single_result( -SQL => $sqlFamilies );
```

Comment:
Place the filter for `default` here instead of further down - no point in running this query when we don't need to

Reply:
Actually, I prefer the type of structure you've got further down ( ~ `desc10`) where you're defining the SQL and description, and running the query and datacheck in one block, rather than the grouping here (all SQL defined, all queries run, etc)

Reply:
This is also my preferred structure :+1:
I am moving the family and other 'default' only collection queries outside of the loop now, and hopefully all should become clearer, and more importantly logical!